lib/seattleflu/db/cli/command/etl/presence_absence.py

```diff
@@ -0,0 +1,385 @@
+    flag_value  = "prompt")
+
+@click.option("--commit", "action",
+    help        = "Save changes to the database",
+    flag_value  = "commit")
+
+def etl_presence_absence(*, action: str):
+    LOG.debug(f"Starting the presence_absence ETL routine, revision {REVISION}")
+
+    db = DatabaseSession()
+
+    # Fetch and iterate over presence-absence tests that aren't processed
+    #
+    # Rows we fetch are locked for update so that two instances of this
+    # command don't try to process the same presence-absence tests.
+    LOG.debug("Fetching unprocessed presence-absence tests")
+
+    presence_absence = db.cursor("presence_absence")
+    presence_absence.execute("""
+        select presence_absence_id as id, document
+          from receiving.presence_absence
+         where not processing_log @> %s
+         order by id
+           for update
+        """, (Json([{ "revision": REVISION }]),))
+
+    processed_without_error = None
+
+    try:
+        for group in presence_absence:
+            with db.savepoint(f"presence_absence group {group.id}"):
+                LOG.info(f"Processing presence_absence group {group.id}")
+
+                for received_sample in group.document["store"]["items"]:
+
+                    # Eventually, we will want to convert the external identifier
+                    # (e.g. barcode) to the original uuid.
+                    sample = find_or_create_sample(db,
+                        identifier  = sample_identifier(db,received_sample["investigatorId"]), 
+                        details = sample_details(received_sample)
+                        # encounter_id = TODO,
+                    )
+
+                    for test_result in received_sample["targetResults"]:
+                        # Most of the time we expect to see existing targets so a
+                        # select-first approach makes the most sense to avoid useless
+                        # updates.
+                        target = find_or_create_target(db,
+                            identifier = test_result["geneTarget"],
+                            control_status = target_control_status(test_result["controlStatus"], test_result["geneTarget"]))
+
+                        # Most of the time we expect to see new samples and new
+                        # presence_absence tests, so an insert-first approach makes more sense.
+                        # Presence-absence tests we see more than once are presumed to be
+                        # corrections.
+                        upsert_presence_absence(db,
+                            identifier = test_result["id"],
+                            sample_id  = sample.id,
+                            target_id  = target.id,
+                            present    = get_target_result(test_result["targetStatus"], 
+                                                           sample.identifier, target.identifier),
+                            well_details = test_result["wellResults"])
```

Comment:
We will soon have other presence/absence results tested by a different method (Cepheid) which doesn't report the concept of wells but will have extra details we'll want to save. I think we should rename the `well_details` column to just `details` (in keeping with our convention elsewhere) and make the document here be:

```python
{
    "replicates": test_result["wellResults"],
}
```